server/internal/core/file/service.go

```diff
@@ -0,0 +1,40 @@
+package file
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/oxidrive/oxidrive/server/internal/core/user"
+	"github.com/rs/zerolog"
+)
+
+type Service struct {
+	filesContent  FilesContent
+	filesMetadata FilesMetadata
+}
+
+func NewService(filesContent FilesContent, filesMetadata FilesMetadata) Service {
+	return Service{
+		filesContent:  filesContent,
+		filesMetadata: filesMetadata,
+	}
+}
+
+func (s *Service) Upload(ctx context.Context, u user.User, content Content, path Path, size Size, logger zerolog.Logger) error {
```

Comment:
Could the content, path, and size params be put in a `FileUpload` struct or something similar? Functions with more than 4 arguments are hard to visually parse

Reply:
Also, it's better to put the logger instance inside the `Service` struct than to pass it as an arg (and soon we will actually get rid of it completely in favour of domain probes)

Reply:
If we remove the `user` and the `logger`, arguments are reduced to three (ctx does not count since it's basically in any function in Go 🙃 ). I would keep them as arguments and not as an additional struct to avoid to jump back and forth only to understand what the function needs

Reply:
We do need the user here to know who is the owner of the file tho